libc-bottom-half/cloudlibc/src/libc/unistd/symlinkat.c

```diff
@@ -10,8 +10,7 @@
 #include <unistd.h>
 
 int __wasilibc_nocwd_symlinkat(const char *path1, int fd, const char *path2) {
-  __wasi_errno_t error =
-      __wasi_path_symlink(path1, strlen(path1), fd, path2, strlen(path2));
+  __wasi_errno_t error = __wasi_path_symlink(path1, fd, path2);
```

Comment:
Given that these are now library calls its makes me wonder if we should continue to use the `__` prefix.   Perhaps these should be just `wasi_path_symlink` and then the real imports could be `__wasi_path_symlink`?    Would most likely mean changing all the types too.

Reply:
I'm happy to implement either way, but @sunfishcode would probably be best to answer the question (I'm not familiar enough with C idioms to know what best to do here).

Reply:
The `__` prefix continues to make sense :-). For example, users can call WASI APIs directly, and may have their own `wasi_*` wrappers around things, so the `__` prefix here ensures that even if they do that, they can also link in libc at the same time.

Reply:
That said, if we wanted to make it easier for people to write code that calls directly into WASI, one thing we could do would be to provide weak aliases named `wasi_*` that point to the `__wasi_` symbols.

Reply:
I just think maybe the `__` prefix is not the right way to carve of our own namespace.   Of course this is unrelated to this PR, but I think `__` is normally reserved for toolchain internal stuff isn't it?  

Again, unrelated to this PR.

Reply:
The C standard just says "reserve for any use". In practice, musl, glibc, and other libc impls commonly use `__` for their own purposes.